### PR TITLE
Refactors Dance Machine + deletes a bunch of sound files

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: The disco machine no longer has a fixed list of songs. It will instead allow players to pick music from a config folder managed by the host and head admins.
add: There is now a bartender jukebox variant of the disco machine that plays music, but has no lights or dancing. This variant will be added the station maps once the feature freeze is over.
/:cl:

The disco machine now lets players pick from a list of songs that are hosted serverside.

Why: Because we do not have the rights to use any of the music that Robustin added when he added this machine. 

Rather than nuking a fairly popular feature I thought converting it into something else that has often been requested would be the path forward that makes everyone happiest. DJ sound board function is gone, I'm pretty sure we need to delete all those sounds as well, and I don't know of a clean way to make it work with both variants.

In the spirit of the freeze I have not added the jukebox to the maps, but if splitting the machine into a dance/non dance variant is still too much of a stretch under what is allowed, I can reopen this when we're accepting feature PRs again.

### CODE REVIEW

**Comment by CitrusGender** [comment]
I'm down. I have no problem with this, perhaps we can get some good music for the server now.

**Comment by Shadowflame909** [comment]
will citrus accept all my Monogatari openings if I try to request such music when they're online?

**Comment by Durkel** [comment]
Good change, and it sets the ground work for getting jukebox in the bar as well.

**Comment by Dorsisdwarf** [comment]
Wait why did you have to remove the DJ soundboard? that was the best bit

**Comment by Spaceinaba** [comment]
maybe dumb idea but you could allow people that are not the bartender to change the song if they input a coin like a real jukebox

**Comment by KorPhaeron** [comment]
@spaceinaba

Jukebox will have to be behind the counter anyway so people dont unwrench it and run every round

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -79,49 +74,40 @@
...
 	if(!allowed(user))
-		to_chat(user,"<span class='warning'>Error: Access Denied - Message: Only the engineering department can be trusted with this kind of power.</span>")
+		to_chat(user,"<span class='warning'>Error: Access Denied.</span>")
 		user.playsound_local(src,'sound/misc/compiler-failure.ogg', 25, 1)
 		return
 	if(!Adjacent(user) && !isAI(user))
 		return
+	if(!songs.len)
+		to_chat(user,"<span class='warning'>Error: No music tracks have been authorized for your station. Petition Central Command to resolve this issue.</span>")
+		user.playsound_local(src,'sound/misc/compiler-failure.ogg', 25, 1)
```
Comment (Cyberboss) [inline code/game/machinery/dance_machine.dm]: just playsound should be fine

File: code/game/machinery/dance_machine.dm [hunk 2]
```diff
@@ -446,30 +414,28 @@
...
 		if(!L || !L.client)
 			continue
 		L.stop_sound_channel(CHANNEL_JUKEBOX)
 	rangers = list()
 
+/obj/machinery/jukebox/disco/dance_over()
+	..()
+	for(var/obj/item/device/flashlight/spotlight/SL in spotlights)
+		qdel(SL)
+	spotlights.Cut()
```
Comment (Cyberboss) [inline code/game/machinery/dance_machine.dm]: QDEL_LIST

File: code/game/machinery/dance_machine.dm [hunk 3]
```diff
@@ -446,30 +414,28 @@
...
+	for(var/obj/effect/overlay/sparkles/SP in sparkles)
+		qdel(SP)
+	sparkles.Cut()
```
Comment (Cyberboss) [inline code/game/machinery/dance_machine.dm]: QDEL_LIST

**Review by Cyberboss — DISMISSED** [review]

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (KorPhaeron) [reply code/game/machinery/dance_machine.dm]: @Robustin Was it on purpose that only the user will recieve the error noise?

**Pull request merged**